### PR TITLE
sway: add support for XDG autostart using systemd

### PR DESCRIPTION
**Pull request by Scrumplex**
### Description

Using the option `wayland.windowManager.sway.systemd.xdgAutostart`, users can now choose to start applications present in `$XDG_CONFIG_HOME/autostart` when starting their sway session.

This change also renames `wayland.windowManager.sway.systemdIntegration` to `wayland.windowManager.sway.systemd.enable`;

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by Scrumplex** [comment]
I am going to pretend to be ofborg and CC @alexarice @sumnerevans @sebtm @oxalica

**Comment by SebTM** [comment]
Should we really switch the default for systemdIntegration from isLinux (so true for many users) to default false? Otherwise looks fine, thanks for the contribution 👍🏻

File: tests/modules/services/window-managers/sway/sway-systemd-autostart.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    # overriding findutils causes issues
+    config.menu = "${pkgs.dmenu}/bin/dmenu_run";
+
+    systemd.xdgAutostart = true;
+  };
+
+  nmt.script = ''
+    assertFileExists home-files/.config/systemd/user/sway-session.target
+    assertFileContent home-files/.config/systemd/user/sway-session.target \
+      ${./sway-systemd-autostart.target}
```
Comment (SebTM) [inline tests/modules/services/window-managers/sway/sway-systemd-autostart.nix:18]: Shound't it be `home-files/.config/systemd/user/sway-systemd-autostart.target` ?

**Comment by SebTM** [comment]
I only found [this](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/system/boot/systemd/user.nix#L40) reference as I searched the service definition for xdg-desktop-autostart, so I'm confident it comes somehow with enabled systemd-support in NixOS-module but would be interested in clarification ✌🏻

**Comment by Scrumplex** [comment]
That target and the accompanying generator are part of systemd. As you pointed out, the target is installed as part of the systemd NixOS module. I am confident that it is also shipped on every other distribution, as DEs like KDE Plasma or GNOME actually use that target to autostart apps, when they use systemd as their init system (which is optional in the case of Plasma)

**Comment by Scrumplex** [comment]
> Should we really switch the default for systemdIntegration from isLinux (so true for many users) to default false

I didn't change the default. You might mistake the example value I gave as a new default

**Review by SebTM — APPROVED** [review]

**Comment by SebTM** [comment]
@sumnerevans @alexarice Ping, can we get this merged? :)

File: modules/services/window-managers/i3-sway/sway.nix [hunk 2]
```diff
@@ -469,13 +491,16 @@ in {
...
       };
 
-      systemd.user.targets.sway-session = mkIf cfg.systemdIntegration {
+      systemd.user.targets.sway-session = mkIf cfg.systemd.enable {
         Unit = {
           Description = "sway compositor session";
           Documentation = [ "man:systemd.special(7)" ];
           BindsTo = [ "graphical-session.target" ];
-          Wants = [ "graphical-session-pre.target" ];
+          Wants = [ "graphical-session-pre.target" ]
```
Comment (alexarice) [inline modules/services/window-managers/i3-sway/sway.nix:499]: Do we need both the `Wants` and `Before` field?

**Review by alexarice — APPROVED** [review]
Seems reasonable though I don't really know anything about systemd units so will leave this for someone more knowledgable to merge.

File: modules/services/window-managers/i3-sway/sway.nix (hunk 2 above)
Comment (Scrumplex) [reply modules/services/window-managers/i3-sway/sway.nix:499]: Yes. Putting it into `Wants` is required for the target to start at all, and putting it into `Before` ensures that it only gets started *after* `sway-session.target` is considered active.

This should avoid any race conditions that might arise.

Comment (SebTM) [reply modules/services/window-managers/i3-sway/sway.nix:499]: @alexarice Who could be this someone? Can we ask someone to have a look? On my try it worked fine but I'm also not an expert in systemd-units neither have merge permissions :)

**Comment by ncfavier** [comment]
I certainly don't understand systemd units, but people seem happy so I'll take a shot in the dark and merge.

**Pull request merged**